autogpt_platform/frontend/src/lib/autogpt-server-api/types.ts

```diff
@@ -65,6 +65,167 @@ export type BlockIOSimpleTypeSubSchema =
   | BlockIOBooleanSubSchema
   | BlockIONullSubSchema;
 
+export enum DataType {
+  SHORT_TEXT = "short-text",
+  LONG_TEXT = "long-text",
+  NUMBER = "number",
+  DATE = "date",
+  TIME = "time",
+  DATE_TIME = "date-time",
+  FILE = "file",
+  SELECT = "select",
+  MULTI_SELECT = "multi-select",
+  TOGGLE = "toggle",
+  CREDENTIALS = "credentials",
+  OBJECT = "object",
+  KEY_VALUE = "key-value",
+  ARRAY = "array",
+}
+
+export function determineDataType(schema: BlockIOSubSchema): DataType {
+  function handleStringSchema(strSchema: BlockIOStringSubSchema): DataType {
+    if ("format" in strSchema && strSchema.format) {
+      switch (strSchema.format) {
+        case "date":
+          return DataType.DATE;
+        case "time":
+          return DataType.TIME;
+        case "date-time":
+          return DataType.DATE_TIME;
+        case "file":
+          return DataType.FILE;
+        case "short-text":
+          return DataType.SHORT_TEXT;
+        case "long-text":
+          return DataType.LONG_TEXT;
+        default:
+          break;
+      }
```

Comment:
I'm pretty sure TS has a better way to do this 😅

Reply:
like by using an object ? Do you think it's a better way? Seems like a switch case is built literally for this purpose.

Reply:
Yeah I'd either use the map properties of `DataType` itself (`if (strSchema.format in DataType) return strSchema.format as DataType`) or make an explicit map of string value to `DataType`. A `switch` like this is *suitable* but since all it does is map a set of values to a different set of values, an object seems like the better choice here.

Reply:
> if (strSchema.format in DataType) return strSchema.format as DataType

Unfortunately, we can't do this because the pedantic format uses the kebab case, while the enum uses the snake case.

> A switch like this is suitable but since all it does is map a set of values to a different set of values, an object value2 = { ...mapping }[value1] seems like the better (for DX) choice here.

Have you checked the rest of the logic below? also this is just a helper of another function, so this matching is just a small part of the bigger logic. Though I updated it using a map now.